frontend/src/concepts/pipelines/topology/parseUtils.ts

```diff
@@ -159,24 +159,131 @@ export const parseRuntimeInfo = (
     startTime: thisTaskDetail[0].start_time,
     completeTime: thisTaskDetail[0].end_time,
     state: lowestProgress(thisTaskDetail),
-    taskId: thisTaskDetail[0].task_id,
+    taskId: `task.${taskId}`,
     podName: thisTaskDetail[0].child_tasks?.find((o) => o.pod_name)?.pod_name,
   };
 };
 
-export const translateStatusForNode = (stateKF?: RuntimeStateKF): RunStatus | undefined => {
-  switch (stateKF) {
+export const parseRuntimeInfoFromExecutions = (
+  taskId: string,
+  executions: Execution[] | null,
+): PipelineTaskRunStatus | undefined => {
+  if (!executions) {
+    return undefined;
+  }
+
+  const execution = executions.find(
+    (e) => e.getCustomPropertiesMap().get('task_name')?.getStringValue() === taskId,
+  );
+
+  if (!execution) {
+    return undefined;
+  }
+
+  const lastUpdatedTime = execution.getLastUpdateTimeSinceEpoch();
+  let completeTime;
+  if (
+    lastUpdatedTime &&
+    (execution.getLastKnownState() === Execution.State.COMPLETE ||
```

Comment:
nit: can we add a constant for `execution.getLastKnownState()`?

Reply:
Also, what sets these states apart from the rest? Maybe they should be stored in their own array, or at least we can add a comment as to why these states are chosen?

Reply:
The states are from here https://github.com/opendatahub-io/data-science-pipelines/blob/master/frontend/src/components/tabs/RuntimeNodeDetailsV2.tsx#L245-L253, I will add a comment to it.